CHANGES.md

```diff
@@ -13,14 +13,35 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 - A small tutorial in tests/OK/tuto.lp.
 
+#### Type checker with refinement
+
+Implemented a refiner, a type checker which is able to edit the type checked
+term. The algorithm is taken from <https://arxiv.org/abs/1202.4905>.
+
+- New type checker (with refinement)
+- Added placeholders. Placeholders are the holes in the concrete syntax, they
+  are transformed into metavariables by the type checker. It simplifies some
+  code related to metavariables. *Placeholders cannot appear nonlinearly in a
+  term*.
```

Comment:
Thanks to provide some justification with a reference.

Reply:
For the nonlinearity?

Reply:
Yes, please. And for removing the input of metavariable applications by users too.

Reply:
I've quoted the article in the changelog (which is the top comment of this PR for now)

Reply:
Right but it would be nice to be more precise and refer to some specific places in the paper so that one does not have to read all the paper to understand why.